lib/magicbell/api_operations.rb

```diff
@@ -34,11 +36,19 @@ def default_headers
 
     def raise_http_error_unless_2xx_response(response)
       return if response.success?
-
+      
       e = MagicBell::Client::HTTPError.new
       e.response_status = response.code
       e.response_headers = response.headers.to_h
       e.response_body = response.body
+      unless e.response_body.nil? || e.response_body.empty?
+        body = JSON.parse(response.body)
+        e.errors = body["errors"]
```

Comment:
Would it also be better to have specific error classes? What are the pros and cons?

Reply:
Not sure if we should have specific error classes;
If we have specific error classes, everytime we change / add new error code in the api, the ruby client will have to be updated as well.

Reply:
@gauravmagicbell I was thinking that the specific error classes help people do better error handling in their code. We don't have to work on them now but I think we should soon. 

We can always default to the more generic error classes if we don't have a specific one.